docs/gallery/howto/autogen/aggregate.py

```diff
@@ -0,0 +1,292 @@
+"""
+==================================
+Aggregate data from multiple tasks
+==================================
+"""
+# %%
+# Introduction
+# ============
+# In this tutorial, you will learn how to aggregate data from the outputs of
+# multiple tasks by linking tasks or by using context. Then we discuss at the
+# end how to deal with nested datatypes.
+#
+# By default, the workgraph only allows one link per input socket, mirroring
+# typical function argument behavior where each argument accepts a single
+# value. However, for scenarios requiring dynamic input port handling, e.g.,
+# functions with variable keyword arguments, the link limit can be safely
+# expanded to accommodate multiple inputs. An input port can be marked as dynamic
+# by using the double asterisk symbol `**` as it is also done to mark an
+# arbitrary number of keyword arguments. This adjustment is particularly
+# useful in tasks designed to aggregate or process collections of data.
+
+
+# Load the AiiDA profile.
+from aiida import load_profile
+
+load_profile()
+
+# %%
+# Dynamic input ports vs orm.Dict
+# --------------------------
+# We want to first focus shortly on the difference between a dynamic input ports and a orm.Dict as input
+# as this is essential for understanding the aggregation mechanism. For that we firstly
+# imitate the `aggregate` function using a Dict as input.
+
+from aiida.orm import Int, Dict
+from aiida_workgraph.utils import generate_node_graph
+from aiida_workgraph import task, WorkGraph
+
+
+@task.calcfunction()
+def aggregate(
+    **collected_values,
+):  # We use the double asterisk to mark it as an dynamic input port
+    for key, value in collected_values.items():
+        print("key:", key, "value:", value)
+    return Int(sum(collected_values.values()))
+
+
+@task.calcfunction
+def aggregate_dict(
+    collected_values,
+):  # We use the double asterisk to mark it as an dynamic input port
+    for key, value in collected_values.items():
+        print("key:", key, "value:", value)
+    return Int(sum(collected_values.get_dict().values()))
+
+
+some_dict = {f"value{i}": Int(i) for i in range(3)}
+aggregate_sum = aggregate(**some_dict)
+aggregate_dict_sum = aggregate_dict(Dict(some_dict))
```

Comment:
Add a note to let the user know that this is not the correct way to use Dict. One should not use an `orm.Data` inside an `orm.Dict`.

Reply:
```suggestion
# Note that it is generally not recommended to have nested orm.Data types like in this case orm.Int in an orm.Data
# This will only work for json serializable types as orm.Int in this case. You will see in the provenance graph later
# that the information about the orm.Int's is completely lost. 
aggregate_dict_sum = aggregate_dict(Dict(some_dict))
```